Model/Api.php

```diff
@@ -479,6 +496,135 @@ function lookupTaxes($itemsByType, $shippingAssignment, $quote)
+            
+            // Check if fallback to Magento is enabled
+            if ($this->_isFallbackToMagentoEnabled()) {
+                $this->_tclogger->info('TaxCloud lookup returned error response, falling back to Magento tax rates');
+                return $this->getMagentoTaxRates($itemsByType, $shippingAssignment, $quote);
+            }
+            
+            return $result;
+        }
+    }
+
+    /**
+     * Get Magento's default tax rates for fallback when TaxCloud fails
+     * @param $itemsByType
+     * @param $shippingAssignment
+     * @param $quote
+     * @return array
+     */
+    private function getMagentoTaxRates($itemsByType, $shippingAssignment, $quote)
+    {
+        $this->_tclogger->info('Falling back to Magento tax rates');
+        
+        $result = array(self::ITEM_TYPE_PRODUCT => array(), self::ITEM_TYPE_SHIPPING => 0);
+        
+        $address = $shippingAssignment->getShipping()->getAddress();
+        if (!$address) {
+            return $result;
+        }
+        
+        // Get the tax calculation service from the Tax model
+        $taxCalculationService = $this->_objectFactory->create(\Magento\Tax\Api\TaxCalculationInterface::class);
+        
+        if (!$taxCalculationService) {
+            $this->_tclogger->info('Could not get Magento tax calculation service');
+            return $result;
+        }
+        
+        try {
+            // Create quote details for tax calculation
+            $quoteDetailsFactory = $this->_objectFactory->create(\Magento\Tax\Api\Data\QuoteDetailsInterfaceFactory::class);
+            $quoteDetailsItemFactory = $this->_objectFactory->create(\Magento\Tax\Api\Data\QuoteDetailsItemInterfaceFactory::class);
+            $taxClassKeyFactory = $this->_objectFactory->create(\Magento\Tax\Api\Data\TaxClassKeyInterfaceFactory::class);
+            $customerAddressFactory = $this->_objectFactory->create(\Magento\Customer\Api\Data\AddressInterfaceFactory::class);
+            $customerAddressRegionFactory = $this->_objectFactory->create(\Magento\Customer\Api\Data\RegionInterfaceFactory::class);
+            
+            if (!$quoteDetailsFactory || !$quoteDetailsItemFactory || !$taxClassKeyFactory || 
+                !$customerAddressFactory || !$customerAddressRegionFactory) {
+                $this->_tclogger->info('Could not create required factories for Magento tax calculation');
+                return $result;
+            }
+            
+            // Build customer address for tax calculation
+            $customerAddress = $customerAddressFactory->create();
+            $customerAddress->setCountryId($address->getCountryId());
+            $customerAddress->setRegionId($address->getRegionId());
+            $customerAddress->setPostcode($address->getPostcode());
+            $customerAddress->setCity($address->getCity());
+            $customerAddress->setStreet($address->getStreet());
```

Comment:
Not sure it's an "improvement", but would it make sense to have a `setFromAddress` method that does all this internally?